cmd/k8s-operator/connector.go

```diff
@@ -211,7 +211,30 @@ func (a *ConnectorReconciler) maybeProvisionConnector(ctx context.Context, logge
 	gaugeConnectorResources.Set(int64(connectors.Len()))
 
 	_, err := a.ssr.Provision(ctx, logger, sts)
-	return err
+	if err != nil {
+		return err
+	}
+
+	_, tsHost, ips, err := a.ssr.DeviceInfo(ctx, crl)
+	if err != nil {
+		return err
+	}
+
+	if tsHost == "" {
+		logger.Debugf("no Tailscale hostname known yet, waiting for connector pod to finish auth")
+		// No hostname yet. Wait for the connector pod to auth.
+		cn.Status.TailnetIPs = nil
+		cn.Status.Hostname = ""
+		if err := a.Status().Update(ctx, cn); err != nil {
+			return fmt.Errorf("failed to update connector status: %w", err)
```

Comment:
Can we rely on the caller setting status (i.e https://github.com/tailscale/tailscale/blob/main/cmd/k8s-operator/connector.go#L153) instead of doing it here as well?

Reply:
Oops, good point, addressed in 3a0b51e